File: Terceiro periodo/Performance Ciberfisico/ex1.py
import time

def main():
    # Número de caracteres a serem impressos
    num_chars = 5000000
    
    print(f"Iniciando impressão de {num_chars:,} caracteres 'X'...")
    
    # Marca o tempo de início
    start_time = time.time()
    
    # Imprime os caracteres "X"
    for i in range(num_chars):
        print("X", end="")
    
    # Marca o tempo de fim
    end_time = time.time()
    
    # Calcula o tempo decorrido
    elapsed_time = end_time - start_time
    
    print(f"\n\nTempo decorrido: {elapsed_time:.4f} segundos")
    print(f"Caracteres por segundo: {num_chars / elapsed_time:.0f}")

if __name__ == "__main__":
    main()

File: Terceiro periodo/Performance Ciberfisico/ex1.c
#include <stdio.h>
#include <time.h>

int main() {
    // Número de caracteres a serem impressos
    const int num_chars = 5000000;
    
    printf("Iniciando impressão de %d caracteres 'X'...\n", num_chars);
    
    // Marca o tempo de início
    clock_t start_time = clock();
    
    // Imprime os caracteres "X"
    for (int i = 0; i < num_chars; i++) {
        printf("X");
    }
    
    // Marca o tempo de fim
    clock_t end_time = clock();
    
    // Calcula o tempo decorrido em segundos
    double elapsed_time = ((double)(end_time - start_time)) / CLOCKS_PER_SEC;
    
    printf("\n\nTempo decorrido: %.4f segundos\n", elapsed_time);
    printf("Caracteres por segundo: %.0f\n", num_chars / elapsed_time);
    
    return 0;
}
